src/cljs/vanilla/widgets/stoplight_widget.cljs

```diff
@@ -0,0 +1,50 @@
+(ns vanilla.widgets.stoplight-widget
+  (:require [reagent.core :as r :refer [atom]]
+            [vanilla.widgets.basic-widget :as basic]))
+
+
+(defn- stoplight [id val]
+
+  ;(.log js/console (str "stoplight " id ", " val))
+
+  (let [c (condp = val
+            :up "green"
+            :warning "yellow"
+            :fault "red")]
+    ^{:key id}
+    [:button.button {:style    {:background-color c
+                                :width            "20%"
+                                :border-spacing   "5px"}
+                     :on-click #(.log js/console (str "clicked " id))}
+     id]))
+
+
+(defn- stoplight-row [data]
+
+  ;(.log js/console (str "stoplight-row " data))
+
+  ^{:key (str data)}
+  [:buttons
+   (for [[k v] data]
+     (doall
+
+       ;(.log js/console (str "stoplight-row> " k "-" v))
+
+       (stoplight k v)))])
+
+
+
+(defn make-widget [name data options]
+
+  (prn ":stoplight make-widget" name
```

Comment:
comment these out (prn ...) blocks now that things are working, but leave the commented code so we can turn it back on in the future